test/arithmetic.jl

```diff
@@ -28,74 +28,204 @@
+        @test minimum(Arf[10:20; 0:9]) == Arf(0)
+        @test maximum(Arf[10:20; 0:9]) == Arf(20)
+        @test extrema(Arf[10:20; 0:9]) == (Arf(0), Arf(20))
     end
 
-    @testset "midref" begin
-        x = Arb(0.25)
-        @test Arblib.midref(x) isa ArfRef
-        @test startswith(sprint(show, x), "0.250")
-        @test Float64(Arblib.midref(x)) isa Float64
-        @test Float64(Arblib.midref(x)) == 0.25
-        @test Float64(x) == 0.25
-        @test sprint(show, x) == sprint(show, x[])
+    @testset "$T" for T in [Arb, Acb]
+        @test T(1) + T(2) ==
+              T(1) + 2 ==
+              2 + T(1) ==
+              T(1) + UInt(2) ==
+              UInt(2) + T(1) ==
+              T(1) + UInt8(2) ==
+              UInt8(2) + T(1) ==
+              T(3)
+        @test T(1) - T(2) == T(1) - 2 == T(1) - UInt(2) == T(-1)
+        @test T(2) * T(3) ==
+              T(2) * 3 ==
+              3 * T(2) ==
+              T(2) * UInt(3) ==
+              UInt(3) * T(2) ==
+              T(6)
+        @test T(6) / T(2) == T(6) / 2 == T(6) / UInt(2) == T(3)
+
+        @test Arblib.root(T(8), 3) == T(2)
+
+        for f in [
+            inv,
+            sqrt,
+            log,
+            log1p,
+            exp,
+            expm1,
+            sin,
+            cos,
+            tan,
+            cot,
+            sec,
+            csc,
+            atan,
+            asin,
+            acos,
+            sinh,
+            cosh,
+            tanh,
+            coth,
+            sech,
+            csch,
+            atanh,
+            asinh,
+        ]
+            # TODO: Replace with ≈
+            @test abs(f(T(0.5)) - f(0.5)) <= 1e-15
+        end
+        # TODO: Replace with ≈
+        @test abs(acosh(T(2)) - acosh(2)) <= 1e-15
+
+        @test Arblib.rsqrt(T(1 // 4)) == T(2)
+        @test Arblib.sqr(T(3)) == T(9)
+
+        @test sinpi(T(1)) == T(0)
+        @test cospi(T(1)) == T(-1)
+        @test Arblib.tanpi(T(1)) == T(0)
+        # TODO: Replace with ≈
+        @test abs(Arblib.cotpi(T(0.5))) <= 1e-15
+        @test abs(Arblib.cscpi(T(0.5)) - 1) <= 1e-15
+        @test abs(sinc(T(0.5)) - sinc(0.5)) <= 1e-15
+
+        @test isequal(sincos(T(1)), (sin(T(1)), cos(T(1))))
+        @test isequal(Arblib.sincospi(T(1)), (sinpi(T(1)), cospi(T(1))))
+        @test isequal(Arblib.sinhcosh(T(1)), (sinh(T(1)), cosh(T(1))))
     end
 
-    @testset "radref" begin
-        x = Arb(0.25)
-        m = Arblib.radref(x)
-        @test m isa MagRef
-        m[] = 1.0
-        @test Float64(m) ≥ 1.0
-        @test sprint(show, m) == sprint(show, m[])
+    @testset "Arb - specific" begin
+        @test Arb(1) + Arf(2) == Arf(2) + Arb(1) == Arb(3)
+        @test Arb(1) - Arf(2) == Arb(-1)
+        @test Arb(2) * Arf(3) == Arf(3) * Arb(2) == Arb(6)
+        @test Arb(6) / Arf(2) == UInt(6) / Arb(2) == UInt8(6) / Arb(2) == Arb(3)
+
+        @test Arb(2)^Arb(2) == Arb(2)^UInt(2) == Arb(2)^UInt8(2) == Arb(4)
```

Comment:
also add
```julia
@test Arb(2)^2 == 2^2 == (Arb(2)^-2)^-1
```

Reply:
Done! Well, except for the 2^2, where I just put 4 ^^